### PR TITLE
fix first tap doesn't work when selecting a photo in the gallery

### LINKED ISSUE #18872: [Android] First tap doesn't work when selecting a photo in the gallery
# Bug Report

Causes these e2e to fail:
test_1_1_chat_send_image_save_and_share, id: 703391
test_group_chat_send_image_save_and_share, id: 703297
test_community_one_image_send_reply, id: 702859

### Reproduction

1. Open chat
2. Open gallery
3. Tap a photo

#### Expected behavior
The photo is selected

#### Actual behavior
Nothing happens, selection works only on the second tap

https://github.com/status-im/status-mobile/assets/67952253/9bae0eb1-8285-43c0-8386-3b88a90a092d

### Additional Information

- Status version: nightly 16/02
- Operating System: Android

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes: https://github.com/status-im/status-mobile/issues/18872

status: ready

### CODE REVIEW

File: src/status_im/common/bottom_sheet_screen/view.cljs [hunk 1]
```diff
@@ -87,7 +87,7 @@
            [content
             {:insets           insets
              :close            close
-             :scroll-enabled?  @scroll-enabled?
+             :scroll-enabled?  scroll-enabled?
```
Comment (Parveshdhull) [inline src/status_im/common/bottom_sheet_screen/view.cljs:90]: Bug cause: Early deref rerendered parent and reinitialized states in child components

File: src/status_im/contexts/chat/messenger/photo_selector/view.cljs [hunk 2]
```diff
@@ -70,13 +70,14 @@
...
-                                 (show-photo-limit-toast)
-                                 (swap! selected conj item))))
-      :accessibility-label (str "image-" index)}
+     {:on-press                (fn []
+                                 (if item-selected?
+                                   (swap! selected remove-selected item)
+                                   (if (>= (count @selected) constants/max-album-photos)
+                                     (show-photo-limit-toast)
+                                     (swap! selected conj item))))
+      :allow-multiple-presses? true
```
Comment (Parveshdhull) [inline src/status_im/contexts/chat/messenger/photo_selector/view.cljs:79]: Additional Improvement (more [details](https://discord.com/channels/1103692771585433630/1103692773363810317/1204009899936452608))

File: src/status_im/contexts/chat/messenger/photo_selector/view.cljs [hunk 3]
```diff
@@ -91,8 +92,6 @@
 
 (defn photo-selector
   [{:keys [scroll-enabled? on-scroll current-scroll close] :as sheet}]
-  (rf/dispatch [:photo-selector/get-photos-for-selected-album])
-  (rf/dispatch [:photo-selector/camera-roll-get-albums])
```
Comment (Parveshdhull) [inline src/status_im/contexts/chat/messenger/photo_selector/view.cljs:95]: Moved these calls to event

File: src/status_im/common/bottom_sheet_screen/view.cljs (hunk 1 above)
Comment (mohsen-ghafouri) [reply src/status_im/common/bottom_sheet_screen/view.cljs:90]: Oh so this was the side effect that i was worried about it, my bad. thank you @Parveshdhull

**Review by mohsen-ghafouri — APPROVED** [review]

File: src/status_im/common/bottom_sheet_screen/view.cljs (hunk 1 above)
Comment (J-Son89) [reply src/status_im/common/bottom_sheet_screen/view.cljs:90]: Hmm but passing a ratom is not ideal, no? Then the component unnecessarily needs to be aware that the prop is a ratom

Comment (Parveshdhull) [reply src/status_im/common/bottom_sheet_screen/view.cljs:90]: We can use some naming convention to clarify that the passed prop is ratom.

**Comment by VolodLytvynenko** [comment]
hi @Parveshdhull. thank you for PR. No issues from my side. PR is ready to be merged

**Pull request merged**